packages/apple-pay-integration/src/apple-pay-payment-strategy.ts

```diff
@@ -235,4 +248,27 @@ export default class ApplePayPaymentStrategy implements PaymentStrategy {
             );
         }
     }
+
+    private async _getDataCollector() {
+        const state = await this._paymentIntegrationService.loadPaymentMethod(
+            ApplePayGatewayType.BRAINTREE,
+        );
+
+        const braintreePaymentMethod: PaymentMethod = state.getPaymentMethodOrThrow(
+            ApplePayGatewayType.BRAINTREE,
+        );
+
+        if (!braintreePaymentMethod.clientToken || !braintreePaymentMethod.initializationData) {
```

Comment:
I don't think we should stop user to make a transaction if the error occurs on getting braintree device data. Hence to avoid blocking the flow, it might be better to return empty device data, if there are no `clientToken` or `initializationData`.

Reply:
Thanks. Done